lib/src/imgcodecs/imgcodecs.dart

```diff
@@ -72,10 +72,7 @@ Uint8List imencode(
       ? cvRun(() => cimgcodecs.Image_IMEncode(cExt, img.ref, success, buffer))
       : cvRun(() => cimgcodecs.Image_IMEncode_WithParams(cExt, img.ref, params.ref, success, buffer));
   calloc.free(cExt);
-  if (!success.value) {
-    throw CvException(ErrorCode.StsError.code, msg: "imencode failed, check your params");
-  }
-  return VecUChar.fromPointer(buffer).toU8List();
+  return (success.value, VecUChar.fromPointer(buffer).toU8List());
 }
```

Comment:
Here success is not freed, and maybe manually free VecU8 is better.

```dart
final pSuccess = calloc<ffi.Bool>();
final cExt = ext.toNativeUtf8().cast<ffi.Char>();

  params == null
      ? cvRun(() => cimgcodecs.Image_IMEncode(cExt, img.ref, pSuccess, buffer))
      : cvRun(() => cimgcodecs.Image_IMEncode_WithParams(cExt, img.ref, params.ref, pSuccess, buffer));
  final success = pSuccess.value;
  calloc.free(cExt);
  calloc.free(pSuccess);

  final vec = VecUChar.fromPointer(buffer); // buffer does not need to be freed via calloc.free because it is managed by VecUChar
  final u8List = vec.toU8List(); // will copy data
  vec.dispose();
  return (success, u8List);
```

Actually I think return VecU8 is better, but just keep Uint8List for compatibility.

Reply:
I noticed the success was not freed, the reason I didn't free it is that your didn't, so I thought freeing it will cause problems 😅

Anyway will do these changes at night since I can't do it now

Reply:
It's my fault, when starting this project, I am not very familiar with dart ffi.

> Anyway will do these changes at night since I can't do it now

Then would you mind me to fix it?

Reply:
No I don't mind it, enjoy 

I learned alot of ffi stuff from you , you are very good don't worry we all just forget or miss stuff it happens to the best of us